lumicks/pylake/correlated_stack.py

```diff
@@ -340,6 +340,59 @@ def select_frame(event):
 
         fig.canvas.mpl_connect('button_press_event', select_frame)
 
+    def export_tiff(self, file_name, roi=None):
+        """Export a video of a particular scan plot
+
+        Parameters
+        ----------
+        file_name : str
+            File name to export to.
+        roi : list_like
+            region of interest in pixel values [xmin, xmax, ymin, ymax]
+        """
+        from . import __version__ as version
+
+        def parse_tags(frame):
+            # Parse original file tags into list of tuples [(code, dtype, count, value, writeonce)]
+            # Only tags that are not resolved automatically by `TiffWriter.save()` are needed
+            n_channels = 3 if frame.is_rgb else 1
+            datetime = frame._src.tags["DateTime"].value
+            output_tags = ((274, "H", 1, 1),                            # Orientation
+                           (339, "H", n_channels, (1, )*n_channels),    # SampleFormat
+                           (306, "s", len(datetime), datetime))         # DateTime
+            return output_tags
+
+        # get ROI coordinates
+        if roi is not None:
+            xmin, xmax, ymin, ymax = roi
+        else:
+            ymin, xmin = 0, 0
+            ymax = self._get_frame(0).data.shape[0]
+            xmax = self._get_frame(0).data.shape[1]
+
+        # re-name alignment matrices fields in image description
+        # to reflect the fact that the image has already been processed
+        description = self._get_frame(0)._src.description
+        if description != "":
+            description = json.loads(description)
+            for color in ("blue", "green", "red"):
+                key = f"Alignment {color} channel"
+                description["Applied "+key] = description.pop(key)
```

Comment:
What happens if a user opens a file that was exported in this manner with `CorrelatedStack`?

I guess it wouldn't be aligned again, which is what we want (though they'd be getting warnings all the time). But what if they export that file again, I guess then the alignment matrices could be lost?

Reply:
correct, it's not aligned again because the metdata field names have changed on export. If you try to re-load the exported file, it does indeed issue a warning, but I don't think that's necessarily a bad thing. It reminds the user that they are not working with the original raw data. I fixed it so the metadata is not changed on re-export (the fields remain "Applied alignment color channel")